tests/client/amo/components/TestAddonDetail.js

```diff
@@ -291,19 +298,39 @@ describe('AddonDetail', () => {
         ...fakeAddon,
         type: ADDON_TYPE_THEME,
       },
-      clientSupportsAddons: () => true,
+      getCompatibleClient: () => ({ compatible: true }),
     });
     const button = rootNode.querySelector('.AddonDetail-theme-header-label');
     assert.equal(button.disabled, false);
   });
 
+  it('disables install switch for unsupported clients', () => {
+    const rootNode = renderAsDOMNode({
+      getClientCompatibility: getClientCompatibilityFalse,
+    });
+    assert.isTrue(
+      rootNode.querySelector('.InstallButton-switch input').disabled);
+  });
+
+  it('throws an error if compatibility props are missing', () => {
+    assert.throws(() => {
+      renderAsDOMNode({
+        getClientCompatibility: () => ({
+          compatibility: false,
```

Comment:
When you need to delete props you can't use a spread operator but the code you have here is still susceptible to future test failures when the object signature changes. An easy way to protect against that is this pattern:

````js
// This would be defined at the top of the test.
const incompatibleClientResult = {
  compatible: false,
  maxVersion: null,
  minVersion: null,
  reason: INCOMPATIBLE_NOT_FIREFOX,
};
// Now, in your test you can write this:
const result = { ...incompatibleClientResult };
delete result.minVersion;
assert.throws(
  () => renderAsDOMNode({ getClientCompatibility: () => result },
   /minVersion is required/,
);
````